### PR TITLE
Trees: use Term.ParamClause, not List[Term.Param]

### PR DESCRIPTION
**Pull request by kitbellew**
Helps with #2869.

### CODE REVIEW

**Comment by kitbellew** [comment]
@tgodzik this is the second PR, covering Term.Param clauses.

File: tests/shared/src/test/scala/scala/meta/tests/quasiquotes/ErrorSuite.scala [hunk 1]
```diff
@@ -803,23 +803,6 @@ class ErrorSuite extends FunSuite {
     )
   }
 
-  test("triple-dot mixes with something else in parameter lists") {
```
Comment (tgodzik) [inline tests/shared/src/test/scala/scala/meta/tests/quasiquotes/ErrorSuite.scala:806]: Why did you remove the test? We no longer have this limitation?

**Review by tgodzik — APPROVED** [review]
LGTM! Just one question from me.

File: tests/shared/src/test/scala/scala/meta/tests/quasiquotes/ErrorSuite.scala (hunk 1 above)
Comment (kitbellew) [reply tests/shared/src/test/scala/scala/meta/tests/quasiquotes/ErrorSuite.scala:806]: because it's not broken anymore (it's capable of parsing this quasiquote now). i added a test in success suite instead.
